### ISSUE TITLE
CscToolExe full path with empty CscToolPath no longer works

### ISSUE THREAD

**Issue by nguerrera** [dotnet/roslyn#22478]
In previous versions, if $(CscToolExe) specified a full path to csc.exe, it would be used even if CscToolPath was unset.

However, this no longer works with latest 15.5 bits.

This was discovered when we were unable to ingest a newer Roslyn in to the CLI without also ingesting a change in the way we the compiler location is set.  Now, with dotnet/cli#7311 the CLI no longer relies on the old behavior, so this issue isn't blocking us. However, this could break anyone else who is setting CscToolExe to a full path. There's a [comment](https://github.com/dotnet/roslyn/blob/39633b336bb1cd571174d1ba53fe51db316db5ef/src/Compilers/Core/MSBuildTask/ManagedCompiler.cs#L427-L428) that suggests that it is still supposed to work, but it no longer works in practice.

**Version Used**: 

Microsoft (R) Visual C# Compiler version 2.6.0.62114 (c5ab7f6e)

**Steps to Reproduce**:

1. Install VS 15.3 RTM

2. Install VS 15.5 from dogfood channel SxS

3. Launch  VS 15.5 developer command prompt

4. Set CscToolExe environment variable to point to full path of VS 15.3 RTM compiler
 ```
    set CscToolExe=C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\MSBuild\15.0\bin\Roslyn\csc.exe 
```

5. Build a simple console app in this environment
```
   mkdir somethingsimple
   cd somethingsimple
   dotnet new console
   msbuild somethingsimple.csproj
```

**Expected Behavior**:
The project compiles using the VS 15.3 compiler specified via CscToolExe. 

You can see this working if you reverse the roles of 15.3 and 15.5: Use 15.3 developer command prompt and point CscToolExe at 15.5 csc.exe full path.

**Actual Behavior**:
```
(CoreCompile target) ->
  C:\Program Files (x86)\Microsoft Visual Studio\Preview\Enterprise\MSBuild\15.0\bin\Roslyn\Microsoft.CSharp.Core.targets(84,5): 
  error MSB3082: Task failed because "" was not found. [D:\Src\ttt\ttt.csproj]
```

@khyperia @jaredpar

**Comment by khyperia** [comment]
Just checking - was this _ever_ supposed to work? ( https://github.com/dotnet/cli/pull/7311#discussion_r133783295 ) - CscToolExe _should_ be just the filename, and CscToolPath _should_ be the path, but I'm wondering if we intended support this as a fallback.

**Comment by nguerrera** [comment]
It's quite possible that this was never supposed to work. The more important question is whether any client other than CLI nevertheless depends on it working and whether it is acceptable to break them. 

Note that the current error message is not actionable. I had to read the code to figure out what was happening. So even if we take the stance that this is supposed to be an error, the diagnostic should indicate what the client has done wrong. 

My personal gut feeling is that just supporting it as before is less risky and no more work than making the error message clearer.

**Comment by jcouv** [comment]
Assigned to @khyperia who seems to be working on a fix already.

**Comment by jcouv** [comment]
@nguerrera  This was closed automatically when PR #22487 was merged, but if I read the comments right @khyperia wasn't able to verify end-to-end. 
Would you be able to help with that?

**Comment by khyperia** [comment]
I wasn't able to verify 100% end-to-end, but did figure out (after writing the original description for the PR) how to test a reasonable chunk of end-to-end-ness (which did catch two bugs...)

When the nuget package gets published next (hopefully tomorrow?), hopefully it's easier to test. (I still don't know how to test fully end-to-end, as this original issue is using system-wide msbuild/csc)
